database/database.go

```diff
@@ -137,40 +167,41 @@ func (db *DB) Close() error {
 	return db.staticClient.Disconnect(ctx)
 }
 
-// Ping sends a ping command to verify that the client can connect to the DB and
-// specifically to the primary.
-func (db *DB) Ping(ctx context.Context) error {
-	return db.staticDB.Client().Ping(ctx, readpref.Primary())
-}
-
-// BlockedSkylink fetches the DB record that corresponds to the given skylink
-// from the database.
-func (db *DB) BlockedSkylink(ctx context.Context, s string) (*BlockedSkylink, error) {
-	sr := db.staticSkylinks.FindOne(ctx, bson.M{"skylink": s})
-	if sr.Err() != nil {
-		return nil, sr.Err()
-	}
-	var sl BlockedSkylink
-	err := sr.Decode(&sl)
-	if err != nil {
-		return nil, err
-	}
-	return &sl, nil
-}
-
 // CreateBlockedSkylink creates a new skylink. If the skylink already exists it
 // does nothing.
 func (db *DB) CreateBlockedSkylink(ctx context.Context, skylink *BlockedSkylink) error {
+	// Ensure the hash is set
+	if skylink.Hash == emptyHash {
```

Comment:
You don't need the empty hash helper var I think. `if skylink.Hash == (crypto.Hash{}) {` should work.

Reply:
Updated, I know that works, but I like the helper more because it's just more descriptive and you don't have those ugly brackets.